keps/sig-node/24-apparmor/README.md

```diff
@@ -0,0 +1,794 @@
+
+There are different scenarios in which applying an AppArmor profile may fail,
+below are the ones we mapped and their outcome once this KEP is implemented:
+
+| Scenario                                                                                           | API Server Result    | Kubelet Result                                                                                                                                        |
+| -------------------------------------------------------------------------------------------------- | -------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------- |
+| 1) Using localhost or `runtime/default` profile when container runtime does not support AppArmor.     | Pod created          | The outcome is container runtime dependent. In this scenario containers may 1) fail to start or 2) run normally without having its policies enforced. |
+| 2) Using custom or `runtime/default` profile that restricts actions a container is trying to make. | Pod created          | The outcome is workload and AppArmor dependent. In this scenario containers may 1) fail to start, 2) misbehave or 3) log violations.                  |
+| 3) Using custom profile that does not exist on the node.                                           | Pod created          | Containers fail to start. Retry respecting RestartPolicy and back-off delay.                                                                          |
+| 4) Using an unsupported runtime profile (i.e. `runtime/default-audit`).                            | Pod **not** created. | N/A                                                                                                                                                   |
+| 5) Using localhost profile with invalid name                                                       | Pod **not** created. | N/A                                                                                                                                                   |
+| 6) AppArmor is disabled by the host or the build                                                   | Pod created. | Kubelet puts Pod in blocked state.                                                                                                                    |
+
+Scenario 2 is the expected behavior of using AppArmor and it is included here
+for completeness.
+
+Scenario 5 represents the case of failing the existing validation, which is
+defined at [Pod API](#pod-api).
+
+### Version Skew Strategy
+
+All API skew is resolved in the API server. New Kubelets will only use the
+AppArmor values specified in the fields, and ignore the annotations.
+
+#### Pod Creation
+
+If no AppArmor annotations or fields are specified, no action is necessary.
+
+If the `AppArmor` feature is disabled per feature gate, then the annotations and
+fields are cleared (current behavior).
```

Comment:
Are they really cleared?
Or (what behavior we usually enforce) we forbid setting those fields but the already existing ones remain untouched?

Reply:
The annotation is dropped here: https://github.com/kubernetes/kubernetes/blob/f58f70bd5730658505042cd9baa80f72d3b6e31e/pkg/api/pod/util.go#L526-L532